python_modules/dagster/dagster/_core/definitions/logical_version.py

```diff
@@ -55,6 +71,42 @@ def __new__(
 UNKNOWN_LOGICAL_VERSION: Final[LogicalVersion] = LogicalVersion("UNKNOWN")
 
 
+class LogicalVersionProvenanceSerializer(DefaultNamedTupleSerializer):
```

Comment:
Why do we need a custom serializer?

Reply:
I may ultimately change the format of `LogicalVersionProvenance` to avoid this annoyance, but it requires a custom serializer because it has a dict with `AssetKey` as key (not allowed in JSON).